### ISSUE TITLE
Missing error when return type doesn't match bound TypeVar

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1193]
The following code should emit an error but does not.

```python
from typing import TypeVar

class Animal:
    pass

class Cow(Animal):
    pass

_TA = TypeVar("_TA", bound=Animal)


def fn(p2: _TA) -> _TA:
    p2 = Animal()  # correctly emits error
    return Animal()  # error is not displayed
```

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.88, which I just published. It will also be included in the next release of Pylance.
